### PR TITLE
Hoxhunt integration v2

### PR DESCRIPTION
**Pull request by tomaskukk**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
New version of Hoxhunt integration. Changed support from community from to partner.t

Includes
- Deprecation of old integration
- New integration
- Default playbook
- Incident types
- Incident fields
- Classifier, mapper incoming & outgoing
- Default layout
- Mirroring

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by yaakovpraisler — CHANGES_REQUESTED** [review]
Hey @tomaskukk Thank you for your contribution and sorry for the delay with the review.
Overall it looks good, please see my comments below.

@idovandijk You can start reviewing the non-integration content.

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml
Comment (tomaskukk) [reply Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml:59]: The amount of incidents varies a lot, I think this is a good default, if a customer would have huge amounts of incidents per day, we'll still [apply the max fetch limit](https://github.com/demisto/content/pull/37019/files/d1cf6e062845c8a28ee0eae571fc8f973d70b94b#diff-c58c6b5e7529cb2f06b87629bf7883a0aadaa86799f0679a22f7c558c82b3d28R490) so it's safe even if there are a lot of threats for n days

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml
Comment (tomaskukk) [reply Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml:96]: Good point, I agree with the idea of providing simple ways to use the command. We'll refactor!

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml
Comment (tomaskukk) [reply Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml:201]: I understood that it's optional to add those based on the [docs](https://xsoar.pan.dev/docs/integrations/mirroring_integration#debugging). I didn't find the need to include them. We can add them if necessary, otherwise I'd just skip having them exposed.

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.py
Comment (tomaskukk) [reply Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.py:652]: Not necessarily, but it doesn't seem to affect the incident in any way if the incident is open in Cortex XSOAR. So having it doesn't seem to cause any harm.

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml
Comment (yaakovpraisler) [reply Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml:201]: Ok, I was just wondering because the `get-mapping-fields` is there.

**Comment by idovandijk** [comment]
Thank you @tomaskukk for your patience, I will send my review in the upcoming days.

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.py
Comment (yaakovpraisler) [inline Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.py:338]:
```suggestion
```
When __init__ is not defined, the super.__init__ will be run

File: Packs/Hoxhunt/ReleaseNotes/2_0_0.md
Comment (yaakovpraisler) [inline Packs/Hoxhunt/ReleaseNotes/2_0_0.md:6]:
```suggestion
```
Sorry, I missed this in my previous comment here, I would do it by my self but I don't have permissions.
Please do it for all the RN entries.

**Review by yaakovpraisler — COMMENTED** [review]
Hey @tomaskukk, Let's wait for @idovandijk's review for the incident types, fields etc. then we'll set a call for a demo.

**Review by idovandijk — DISMISSED** [review]
Looks good! I have a few comments regarding the fields:
1. There are many new fields in this PR. While some fields are Hoxhunt specific, there are some fields that could potentially be replaced with existing fields in the marketplace.

Please take a look at the existing incident fields in the CommonTypes pack in the Marketplace.
If you find a suitable replacement field, you can associate it to your Hoxhunt incident types by adding your incident types under **both** the `associatedTypes` **and** under the `systemAssociatedTypes` properties of the field in its JSON file.

If the new field you want to use is a content pack other than CommonTypes, you can decide to keep your existing field, or to use the existing one and add a dependency in your pack to the destination pack. Since using fields from packs other than CommonTypes (which is preinstalled) will introduce a new dependency for your pack, I leave this to you to decide. CommonTypes however, is a Core pack, which means XSOAR users get those fields preinstalled anyway, so it's worth using them.

The advantage to using existing fields rather than adding new ones is that they are shared across different vendors and can be used in widgets, dashboards, etc. For example, if I want to filter all incidents by the "Email To" fields, I won't be seeing incidents where the "Hoxhunt Email To" field is populated.

2. In addition to my previous point regarding fields, there is an additional check I'd like to ask from your side. As the Hoxhunt pack is also available in XSIAM, we can use Field Aliases instead of adding those new alert fields into XSIAM (we will only add them to XSOAR). Please go over the CoreAlertFields pack, and see whether any fields from there can be utilized for your use-case.

For example, let's say the Email Recipient field in the CoreAlertFields pack fulfills our need for our Hox FT Email To field. We will do the following:
A. Add the name of the field Hox FT Email To under the `Aliases` property in the field Email Recipient from the CoreAlertFields pack like so:
image
B. Edit the Hox FT Email To field and add the following property, to make sure this field is added only to XSOAR (because XSIAM will use the alias);
```
"marketplaces": [
"xsoar"
]
```

3. Please rename the fields to have spaces between words in Title Case.

Regarding the playbook: Before using `rasterize` add a task with the `IsIntegrationAvailable` automation to check if Rasterize is available prior to using it.

Once you're able to introduce the changes, I would be happy to see the playbook in action, the mapper and classifier, and the layout :)

If you have any questions please don't hesitate to let me know. Looking forward to get this merged 😎

File: Packs/Hoxhunt/IncidentFields/incidentfield-HoxMessageToUsers.json
Comment (tomaskukk) [reply Packs/Hoxhunt/IncidentFields/incidentfield-HoxMessageToUsers.json:29]: No, I can add `unsearchable: true`, good catch!

**Comment by idovandijk** [comment]
> We also had this chat with @edik24, the reason we wanted to use custom fields is to keep the fields `isReadOnly: true`. These fields should never be edited by the user, as they are kept up to date within our server. User should have no control over them. The suitable replacements that I checked, were `isReadOnly: false`.
> 
> There are a couple of fields that are `isReadOnly: false`, but we didn't find suitable built-in system fields for them.

Thank you for bringing this to my attention. In that case, it makes sense to keep your fields.

> Honestly I'm not that familiar with XSIAM. I trust your guidance, but does it have implications such as mentioned above in the first comment reply (readonly fields)?

Let me get back to you this. As I don't want you to have to make changes and then revert them, please ignore this point for now until I can update with how it will behave.

One more thing, as you make changes to the field names, be aware that these changes will require also updating the mapper and layout to reflect the new names.

**Comment by tomaskukk** [comment]
> Let me get back to you this. As I don't want you to have to make changes and then revert them, please ignore this point for now until I can update with how it will behave.

Alright thank you!

> One more thing, as you make changes to the field names, be aware that these changes will require also updating the mapper and layout to reflect the new names.

Yes! We renamed the field names already and updated the incoming mapper. Outgoing mapper and layouts are using the `cliName` which stayed the same with the changes.

Thank you for all the help so far! Let us know if there's any other changes we need to make out before scheduling the demo :)

**Comment by tomaskukk** [comment]
Hey @idovandijk just wanted to ask if there is something that is expected from us to move this forwards now :)

**Comment by idovandijk** [comment]
> Hey @idovandijk just wanted to ask if there is something that is expected from us to move this forwards now :)

Hi @tomaskukk, it looks like the behavior for field aliases is: the main field's `readOnly` property is propagated to the aliases configured in it. So in your case since your field is set to `readOnly` and the main field in XSIAM is editable, you can also skip that part of my comment :)

We can continue to demo

**Comment by tomaskukk** [comment]
Awesome @idovandijk, would you be able to propose some times that work for you, for example three slots that we could choose from?

**Comment by idovandijk** [comment]
> Awesome @idovandijk, would you be able to propose some times that work for you, for example three slots that we could choose from?

We have 

> Awesome @idovandijk, would you be able to propose some times that work for you, for example three slots that we could choose from?

We've reached out with an email, let us know if that works and if not we can find a new time. Thanks again.

**Comment by tomaskukk** [comment]
I submitted two small commits that:

1. Fixes pack readme not to show contributors twice
2. Fixed a few incident fields that still had wrong name format

**Comment by pennane** [comment]
Reordered the config settings as per previuous demo discussion, also generated the readme and merged from base to resolve conflict.

@yaakovpraisler

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.py
Comment (yaakovpraisler) [inline Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.py:453]: Please use the `headers` and the `headerTransform` function arguments in order to became the headers more readable.
https://xsoar.pan.dev/docs/reference/api/common-server-python#tabletomarkdown

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml
Comment (yaakovpraisler) [inline Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml:78]:
```suggestion
- display: Incident Mirroring Direction
  name: mirror_direction
  defaultvalue: Incoming And Outgoing
  type: 15
  required: false
  additionalinfo: "Choose the direction to mirror the incident: Incoming (from Hoxhunt to Cortex XSOAR), Outgoing (from Cortex XSOAR to Hoxhunt), or Incoming and Outgoing (from/to Cortex XSOAR and Hoxhunt)."
  options:
  - None
  - Incoming
  - Outgoing
  - Incoming And Outgoing
  hidden:
  - marketplacev2
- display: Trust any certificate (not secure)
  name: insecure
  type: 8
  required: false
- display: Use system proxy settings
  name: proxy
  type: 8
  required: false
```
Move the `proxy` to the bottom and removed the "section: Collect" from mirror_direction.
The "section: Collect" should be for related fetch parameters like "IncidentType", "incidentFetchInterval", "first_fetch" etc.

File: Packs/Hoxhunt/ReleaseNotes/2_0_0.json
Comment (yaakovpraisler) [inline Packs/Hoxhunt/ReleaseNotes/2_0_0.json:1]: I don't know why I didn't noticed this before, but you can remove this file because it's not breaking anything, customer that already using the v1 integration will can continue to use it until they're updating to v2.

**Comment by pennane** [comment]
- added explicit section tags into yml (Collect for fetch parameters, Connect for connecting)
- removed release notes __json__ file
- updated create_output headers to be in title case


_example of output:_
CleanShot 2024-12-13 at 16 11 44@2x

**Review by yaakovpraisler — APPROVED** [review]

File: Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml
Comment (yaakovpraisler) [inline Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml:27]:
```suggestion
```
There are 2 time the field `section`

Comment (yaakovpraisler) [inline Packs/Hoxhunt/Integrations/HoxhuntV2/HoxhuntV2.yml:27]:
```suggestion
```
This parameter has 2 times the field `section`, please remove this as it should be in Connect

File: Packs/Hoxhunt/Playbooks/Hox-Enrich-Incident_README.md
Comment (yaakovpraisler) [inline Packs/Hoxhunt/Playbooks/Hox-Enrich-Incident_README.md:37]: The image file doesn't exist under the doc_files dir and hasn't the same file name.
Please move out the image file to the doc_files dir and fix the name and make sure you can see the image in the playbook readme in XSOAR.

**Review by yaakovpraisler — COMMENTED** [review]
I found the following issues during the build failure, please fix them so we can move forward and merge the PR.

Also you need to sign the CLAassistance as described [here](https://github.com/demisto/content/pull/37019#issuecomment-2451783892). (both of you @tomaskukk and @pennane)

Thanks.

**Comment by tomaskukk** [comment]
> I found the following issues in the build pre-commit failure, please fix them so we can move forward and merge the PR.
Also you need to sign the CLAassistance as described https://github.com/demisto/content/pull/37019#issuecomment-2451783892. (both of you @tomaskukk and @pennane)
Thanks.

Done @yaakovpraisler!

**Review by yaakovpraisler — COMMENTED** [review]
The build still fails, you can go to the pre-commit step to see the errors.
Anyway, we need to fix the following -

To fix this:
`Packs/Hoxhunt/Playbooks/Hox-Enrich-Incident.yml: [RM116] - No playbook image found, please add playbook image`
Please add the [.pack-ignore](https://xsoar.pan.dev/docs/packs/packs-format#pack-ignore) file and add the following section:
```
[file:Hox-Enrich-Incident.yml]
ignore=RM116
```

**Comment by pennane** [comment]
Sorry about recurring pre-commit check fails. There are some problems running the pre-commit hook locally so sadly these were not caught before requesting reviews.

But another thing, is upgrading to py 3.12 necessary? We faced some mirroring problems prior when trying to upgrade from 3.10 and that is why it still has been at that version. We can of course check it futher.

**Comment by yaakovpraisler** [comment]
@pennane I didn't notice before but we're not supporting yet 3.12 (this is currently in progress) but I'm not sure this should impact the mirroring.
Anyway, let's upgrade the docker tag to `3.11.10.116949`. (editing my previous comment as well)

**Pull request merged**
